### PR TITLE
[Serve] Implement flask_request and named python request

### PR DESCRIPTION
**Pull request by simon-mo**
- Introduce the concept of `serve.context.web`.
- Use flask_request object instead of ASGI dictionary.
- Make the following interface possible.

image

--------
- Removed `example_rollback.py` as we will remove rollback API soon.

### CODE REVIEW

File: python/ray/experimental/serve/examples/echo_actor.py [hunk 1]
```diff
@@ -1,41 +1,46 @@
 """
-Example actor that adds message to the end of query_string.
+Example actor that adds number to the end of query string.
```
Comment (pcmoritz) [inline python/ray/experimental/serve/examples/echo_actor.py]: Maybe say something like ``that adds an increment to a number``?

File: python/ray/experimental/serve/handle.py [hunk 2]
```diff
@@ -28,11 +30,17 @@ def __init__(self, router_handle, endpoint_name):
         self.router_handle = router_handle
         self.endpoint_name = endpoint_name
 
-    def remote(self, *args):
-        # TODO(simon): Support kwargs once #5606 is merged.
+    def remote(self, *args, **kwargs):
+        if len(args) != 0:
```
Comment (pcmoritz) [inline python/ray/experimental/serve/handle.py]: `#5606` is not merged yet, right? Does this still work?

File: python/ray/experimental/serve/tests/test_api.py [hunk 3]
```diff
@@ -1,33 +1,25 @@
...
-
-# flaky test because the routing table might not be populated
-@flaky(rerun_filter=delay_rerun)
 def test_e2e(serve_instance):
     serve.create_endpoint("endpoint", "/api")
     result = ray.get(
         serve.global_state.kv_store_actor_handle.list_service.remote())
     assert result == {"/api": "endpoint"}
 
+    time.sleep(1)
```
Comment (pcmoritz) [inline python/ray/experimental/serve/tests/test_api.py]: There should probably be a better way to do this :)

**Review by pcmoritz — APPROVED** [review]
Left some small comments (including on the design doc), looks good overall!

File: python/ray/experimental/serve/handle.py (hunk 2 above)
Comment (simon-mo) [reply python/ray/experimental/serve/handle.py]: Yes. We have our own kwargs handling mechanism now. `Queue::enqueue_request` takes in `request_args: List, request_kwargs: Dict` now.

File: python/ray/experimental/serve/tests/test_api.py (hunk 3 above)
Comment (simon-mo) [reply python/ray/experimental/serve/tests/test_api.py]: Fixed.

**Pull request merged**